### PR TITLE
25 Make AdmissionProcess Model 

### PR DESCRIPTION
**Pull request by azizbabul**
[taiga](https://tree.taiga.io/project/hmtanbir-college/us/25?kanban-status=844212)

### CODE REVIEW

File: app/views/admin/admission_processes/_admin_admission_process_image_detail_fields.html.erb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<div class="nested-fields">
+  <div class="field">
+    <%= f.label :details %>
+    <%= f.text_field :details %>
+  </div>
+  <%= link_to_remove_association 'Remove Details', f, class: 'form-button btn btn-default' %>
+</div>
```
Comment (hmtanbir) [inline app/views/admin/admission_processes/_admin_admission_process_image_detail_fields.html.erb]: Add a new line

File: app/views/admin/admission_processes/_admin_admission_process_image_fields.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        <%= f.fields_for :admin_admission_process_image_details do |details_form| %>
+            <%= render 'admin/admission_processes/admin_admission_process_image_detail_fields', f: details_form %>
+        <% end %>
+        <%= link_to_add_association 'Add Details',f,:admin_admission_process_image_details,class: 'btn btn-primary' %>
+      </div>
+    </div>
+
+    <%= link_to_remove_association 'Remove Image', f, class: 'form-button btn btn-default' %>
+  </div>
+</div>
```
Comment (hmtanbir) [inline app/views/admin/admission_processes/_admin_admission_process_image_fields.html.erb]: add a new line

File: db/migrate/20161130152451_create_admin_admission_process_image_details.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class CreateAdminAdmissionProcessImageDetails < ActiveRecord::Migration[5.0]
+  def change
+    create_table :admin_admission_process_image_details do |t|
+      #t.references :admission_process_image, foreign_key: true
+      t.text :details
+      t.references :admission_process_image, foreign_key: true
+      t.timestamps
+    end
+    add_index :admission_process_image_details,[:name]
```
Comment (hmtanbir) [inline db/migrate/20161130152451_create_admin_admission_process_image_details.rb]: How do you get it ?
Please follow story.

File: db/migrate/20161130152451_create_admin_admission_process_image_details.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class CreateAdminAdmissionProcessImageDetails < ActiveRecord::Migration[5.0]
+  def change
+    create_table :admin_admission_process_image_details do |t|
+      #t.references :admission_process_image, foreign_key: true
+      t.text :details
+      t.references :admission_process_image, foreign_key: true
```
Comment (hmtanbir) [inline db/migrate/20161130152451_create_admin_admission_process_image_details.rb]: Please add an index for this foreign key.

**Review by hmtanbir — CHANGES_REQUESTED** [review]
Please fix all comments.

File: db/migrate/20161130152451_create_admin_admission_process_image_details.rb (hunk 3 above)
Comment (azizbabul) [reply db/migrate/20161130152451_create_admin_admission_process_image_details.rb]: fixed -- added in the table

File: db/migrate/20161130152451_create_admin_admission_process_image_details.rb (hunk 4 above)
Comment (azizbabul) [reply db/migrate/20161130152451_create_admin_admission_process_image_details.rb]: it was there , i uncommentd it in line 4 and commented line 6

Comment (hmtanbir) [reply db/migrate/20161130152451_create_admin_admission_process_image_details.rb]: I don't  know why you don't follow my instruction, please fix this error as soon as possible.

File: db/migrate/20161130152451_create_admin_admission_process_image_details.rb (hunk 3 above)
Comment (hmtanbir) [reply db/migrate/20161130152451_create_admin_admission_process_image_details.rb]: Nothing fixed yet, please fix, drop your database and re migration.

File: app/views/admin/admission_processes/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+<p>
+  <strong>Step:</strong>
+  <%= @admin_admission_process.step %>
+</p>
+
+<p>
+  <strong>Name:</strong>
+  <%= @admin_admission_process.name %>
+<% for i in @admin_admission_process.admin_admission_process_images%>
```
Comment (hmtanbir) [inline app/views/admin/admission_processes/show.html.erb]: Use `each` block instead of `for` block

**Review by hmtanbir — CHANGES_REQUESTED** [review]
Please fix all of my comments.

File: db/migrate/20161130152326_create_admin_admission_process_images.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+class CreateAdminAdmissionProcessImages < ActiveRecord::Migration[5.0]
+  def change
+    create_table :admin_admission_process_images do |t|
+      t.references :admission_process, foreign_key: true
```
Comment (hmtanbir) [inline db/migrate/20161130152326_create_admin_admission_process_images.rb]: Please add an index, read story carefully again. If you read it, you will get the process of adding index.

File: db/migrate/20161130152451_create_admin_admission_process_image_details.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+class CreateAdminAdmissionProcessImageDetails < ActiveRecord::Migration[5.0]
+  def change
+    create_table :admin_admission_process_image_details do |t|
+      t.references :admission_process_image, foreign_key: true
```
Comment (hmtanbir) [inline db/migrate/20161130152451_create_admin_admission_process_image_details.rb]: Please add an index, read story carefully again. If you read it, you will get the process of adding index.

File: app/views/admin/admission_processes/_admin_admission_process_image_fields.html.erb [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+        <% end %>
+        <%= link_to_add_association 'Add Details',f,:admin_admission_process_image_details,class: 'btn btn-primary' %>
+      </div>
+
+    </div>
+
+    <%= link_to_remove_association 'Remove Image', f, class: 'form-button btn btn-default' %>
+
+  </div>
+</div>
```
Comment (hmtanbir) [inline app/views/admin/admission_processes/_admin_admission_process_image_fields.html.erb]: add a new line.

**Review by hmtanbir — CHANGES_REQUESTED** [review]

File: db/migrate/20161130152326_create_admin_admission_process_images.rb (hunk 6 above)
Comment (azizbabul) [reply db/migrate/20161130152326_create_admin_admission_process_images.rb]: done

File: db/migrate/20161130152451_create_admin_admission_process_image_details.rb (hunk 7 above)
Comment (azizbabul) [reply db/migrate/20161130152451_create_admin_admission_process_image_details.rb]: done

File: app/views/admin/admission_processes/_admin_admission_process_image_fields.html.erb (hunk 8 above)
Comment (azizbabul) [reply app/views/admin/admission_processes/_admin_admission_process_image_fields.html.erb]: done

File: app/controllers/admin/abouts_controller.rb [hunk 9]
```diff
@@ -1,6 +1,6 @@
 class Admin::AboutsController < ApplicationController
   before_action :set_admin_about, only: [:show, :edit, :update, :destroy]
-
+ layout "college/base"
```
Comment (hmtanbir) [inline app/controllers/admin/abouts_controller.rb]: Why do you add fronted layout in backend ?

File: app/views/admin/admission_processes/_admin_admission_process_image_fields.html.erb [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+        <%= link_to_add_association 'Add Details',f, :admin_admission_process_image_details,class: 'btn btn-primary' %>
+      </div>
+
+    </div>
+
+    <%= link_to_remove_association 'Remove Image',
+                                   f, class: 'form-button btn btn-default' %>
+
+  </div>
+</div>
```
Comment (hmtanbir) [inline app/views/admin/admission_processes/_admin_admission_process_image_fields.html.erb]: Please add a new line.

File: db/migrate/20161130152326_create_admin_admission_process_images.rb [hunk 11]
```diff
@@ -0,0 +1,10 @@
+class CreateAdminAdmissionProcessImages < ActiveRecord::Migration[5.0]
+  def change
+    create_table :admin_admission_process_images do |t|
+      t.references :admission_process, foreign_key: true,
+                   index: {name: :admission_process_image_}
```
Comment (hmtanbir) [inline db/migrate/20161130152326_create_admin_admission_process_images.rb]: Why do you use `underslash` end of the `admission_process_image_` word ?

File: db/migrate/20161130152451_create_admin_admission_process_image_details.rb [hunk 12]
```diff
@@ -0,0 +1,11 @@
+class CreateAdminAdmissionProcessImageDetails < ActiveRecord::Migration[5.0]
+  def change
+    create_table :admin_admission_process_image_details do |t|
+      t.text :details
+      #t.index :name
+      t.references :admission_process_image, foreign_key: true,
+                   index: {name: :admission_process_image_details}
```
Comment (hmtanbir) [inline db/migrate/20161130152451_create_admin_admission_process_image_details.rb:7]: 👍

File: config/routes.rb [hunk 13]
```diff
@@ -1,5 +1,8 @@
 Rails.application.routes.draw do
   namespace :admin do
+    resources :admission_processes
+  end
```
Comment (hmtanbir) [inline config/routes.rb]: Please routes will be like as DRY!

**Review by hmtanbir — CHANGES_REQUESTED** [review]

File: app/controllers/admin/abouts_controller.rb (hunk 9 above)
Comment (azizbabul) [reply app/controllers/admin/abouts_controller.rb]: changed

File: app/views/admin/admission_processes/_admin_admission_process_image_fields.html.erb (hunk 10 above)
Comment (azizbabul) [reply app/views/admin/admission_processes/_admin_admission_process_image_fields.html.erb]: added line

File: db/migrate/20161130152326_create_admin_admission_process_images.rb (hunk 11 above)
Comment (azizbabul) [reply db/migrate/20161130152326_create_admin_admission_process_images.rb]: corrected

File: config/routes.rb (hunk 13 above)
Comment (azizbabul) [reply config/routes.rb]: DRY done

**Review by hmtanbir — APPROVED** [review]

**Pull request merged**